core/src/main/java/org/infinispan/topology/ClusterCacheStatus.java

```diff
@@ -692,6 +694,17 @@ public void setRebalanceEnabled(boolean enabled) {
       }
    }
 
+   /**
+    * Only performs rebalance if the per-cache rebalancing is enabled
+    */
+   public void conditionalRebalance() {
```

Comment:
Ok. One question I have about that is that before checking if rebalancing is enabled, it might invoke `broadcastStableTopologyUpdate`. Is that ok, or should we shuffle things around so that the check comes first ?